components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -506,13 +505,11 @@ class GeckoEngineSession(
 
         override fun onExternalResponse(session: GeckoSession, response: GeckoSession.WebResponseInfo) {
             notifyObservers {
-                val fileName = response.filename
-                    ?: DownloadUtils.guessFileName("", response.uri, response.contentType)
                 onExternalResource(
                         url = response.uri,
                         contentLength = response.contentLength,
                         contentType = response.contentType,
-                        fileName = fileName)
+                        fileName = response.filename)
```

Comment:
Before starting the download, we use the `filename` to show a dialog to [confirm if the user wants to start the download or not](https://github.com/mozilla-mobile/android-components/blob/c6e58ae709f4f3f3284565b317529e9ac390c2f2/components/feature/downloads/src/main/java/mozilla/components/feature/downloads/DownloadsFeature.kt#L94). Could this change affect the name displayed to the user? As we used `DownloadUtils.guessFileName` for edge case where the `filename` wasn't provided.

![Screenshot_20190719-094716](https://user-images.githubusercontent.com/773158/61539890-77558900-aa0a-11e9-9425-15acbe53b9c8.png)

Reply:
Unfortunately the name doesn't match in that case. However, we only don't have the filename for image downloads and the popup will eventually be removed (#2325)

Reply:
👍